### DISCUSSION TITLE
Handle currency differences

### DISCUSSION THREAD

**Discussion by datasmurfen** [ghostfolio/ghostfolio#2759] [Ideas]
Hello, when I buy stocks in USD I add the price pr stock in USD.  My platform Nordnet automatically converts this into my local currency NOK, and my total amount is paid in my local currency.
I have tried to figure out a way to add this, but it seems to me it's calculated back into USD if I change the amount or add only the fee as NOK. 

Concrete example: 
I buy 5 Pepsi Stock for 167,5900 USD.
Exchange rate USD/NOK: 10,975038
NOK: 9196,53 + 49 in fee gives me a total of 9245,53.

**Comment by dtslvr** [comment]
Related to https://github.com/ghostfolio/ghostfolio/issues/2449.

**Reply by brhahlen** [reply]
Also related: 
* #2700 
* #2702

**Reply by brhahlen** [reply]
#2764

**Comment by brhahlen** [comment]
So basically, from what I'm gathering in the different topics, Ghostfolio is not reliable when using any currency other than USD? Even if you have set a different currency as a base?

**Reply by MRobi1** [reply]
Correct. This is due to the fact that ghostfolio converts and stores it's data in USD and then does it's calculations based on today's currency conversion. This causes the original cost basis, as well as all profit/loss calculations to be incorrect.

**Reply by meuter** [reply]
Indeed, I can confirm. This is what I experienced as well.

**Comment by brhahlen** [comment]
Pinging @MRobi1 and @meuter to be added to this discussion.

**Comment by brhahlen** [comment]
Unfortunately, I know next to nothing about TypeScript/Angular, so have no clue how to implement this or where to start even.
However, I do think this is critical to be fixed for wider adoption (and reliable reporting)

**Comment by MRobi1** [comment]
I think the ideal solution would be to store the activities in the selected currency and lock it. So the initial purchase price in the selected currency does not change based on exchange rate. Bonus points if the drop down in activities defaults to the chosen base currency rather than USD, but that's minor.

Then take the current price in USD (or whatever the security trades in), use the current conversion rate to convert to chosen base currency at time of displaying the data, and do all the calculations in the chosen base currency.

Once the initial activity data is stored properly, the calculations are straight forward. The complexity may come with how deep this issue runs. Is it just on the activity pages? Or is the calculation currently done incorrectly throughout the entirety of ghostfolio?

Unfortunately I don't have the skillset to do the coding. But I can certainly beta test and provide input along the way. I feel the same as @brhahlen in that this is critical as it's a core accounting error and does prevent the use of ghostfolio outside of USD.
